test/tools/test_component_tool.py

```diff
@@ -639,3 +639,44 @@ def test_jinja_based_component_tool_in_pipeline(self, monkeypatch):
             result = pipeline.run({"llm": {"messages": [ChatMessage.from_user(text="Hello")], "tools": [tool]}})
 
         assert result["llm"]["replies"][0].text == "A response from the model"
+
+    def test_component_tool_with_super_component_docstrings(self, monkeypatch):
```

Comment:
Let's also add a test for when `input_mapping` contains a input that is forwarded to at least two different components. So one that has something like 
```python
input_mapping = {"text": ["processor.text", "checker.text"]}
```